db/migrate/20150922165007_create_ex_links.rb

```diff
@@ -0,0 +1,15 @@
+class CreateExLinks < ActiveRecord::Migration
+  def change
+    create_table :ex_links do |t|
+      t.string :label
+      t.text :url
+      t.string :tags
```

Comment:
Are you storing the tags in packed format like "tag1,tag2,tag3"? This is complicating your code a lot.

Consider storing them in normalised format instead, i.e. putting tags in a table of their own. Your code will be much cleaner.